src/playback-watcher.js

```diff
@@ -90,6 +92,22 @@ export default class PlaybackWatcher {
     const cancelTimerHandler = () => this.cancelTimer_();
     const fixesBadSeeksHandler = () => this.fixesBadSeeks_();
 
+    const mpc = this.masterPlaylistController_;
+
+    const loaderTypes = ['main', 'subtitle', 'audio'];
+    const loaderChecks = {};
+
+    loaderTypes.forEach((type) => {
+      loaderChecks[type] = {
+        reset: () => this.resetSegmentDownloads_(type),
+        updateend: () => this.checkSegmentDownloads_(type)
+      };
+
+      mpc[`${type}SegmentLoader_`].on('updateend', loaderChecks[type].updateend);
+      mpc[`${type}SegmentLoader_`].on('playlist-update', loaderChecks[type].reset);
```

Comment:
If a rendition switch happens during a playback stall where the buffer isn't changing we want to reset as we cannot assume that the new rendition will also be stalled, until after new appends.